src/main/java/com/microsoft/graph/http/CoreHttpProvider.java

```diff
@@ -417,20 +417,20 @@ public Request authenticateRequest(Request request) {
 
 				final Map<String, String> headers = CoreHttpProvider.getResponseHeadersAsMapStringString(response);
 
+				if(response.body() == null || response.body().contentLength() == 0)
```

Comment:
minor perf: if `body()` is more than a simple property access, consider saving a reference to that into a variable to avoid calling it twice. If it is property access, the overhead should be minimal, so you can ignore this comment in that case.

Reply:
thanks for double checking. I had checking for that and it's simply an accessor to a private field.